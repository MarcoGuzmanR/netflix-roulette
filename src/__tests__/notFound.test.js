import '@testing-library/jest-dom';
import * as React from 'react'
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom'
import NotFound from '../components/movieNotFound/notFound';

test('Shows the not found page message for a wrong movie page', () => {
  const history = createMemoryHistory()
  history.push('/movie-not-found');

  const tree = renderer.create(
    <Router history={history}>
      <NotFound />
    </Router>).toJSON();

  render(
    <Router history={history}>
      <NotFound />
    </Router>
  );

  expect(screen.getByText(/page not found/i)).toBeInTheDocument();
  expect(screen.getByText(/404/i)).toBeInTheDocument();
  expect(screen.getByText(/go back to home/i)).toBeInTheDocument();

  expect(tree).toMatchSnapshot();
});