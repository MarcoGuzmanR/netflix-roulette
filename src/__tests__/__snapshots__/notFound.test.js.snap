// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows the not found page message for a wrong movie page 1`] = `
<div
  className="not-found-container"
>
  <h1>
    Page Not Found
  </h1>
  <h2>
    404
  </h2>
  <button
    className="btn-confirm"
    type="button"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      GO BACK TO HOME
    </a>
  </button>
</div>
`;
